### PR TITLE
Add docs for sharding XCUITest

### PR DESCRIPTION
**Pull request by tianfeng92**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by FriggaHel — APPROVED** [review]

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 1]
```diff
@@ -150,8 +159,87 @@ The next step is to trigger the test by following command:
 ```
 $ saucectl run
 ```
+
 `saucectl` uploads `DemoApp.ipa` and `DemoAppUITests-Runner.ipa` to Sauce VM and executes XCUITest. You can review test results on Sauce UI.
 
+#### Sharding XCUITest
+
+`saucectl` supports executing tests in parallel. To achieve it, `saucectl` needs sharding settings like follows. For more details, check in [yaml configuration](./xcuitest.md#shard).
```
Comment (aliciakrebs) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:167]:
```suggestion
`saucectl` supports running tests in parallel. To tests in parallel, configure `saucectl` as follows. For more details, see [yaml configuration](./xcuitest.md#shard).
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 2]
```diff
@@ -150,8 +159,87 @@ The next step is to trigger the test by following command:
...
+
+```
+suites:
+  - name: "sharded suites"
+    shard: concurrency
+    testListFile: test_classes.txt
+```
+
+The `test_classes.txt` can be generated by following steps.
```
Comment (aliciakrebs) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:176]:
```suggestion
The `test_classes.txt` can be generated by following these steps.
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 3]
```diff
@@ -150,8 +159,87 @@ The next step is to trigger the test by following command:
...
+
+1. Open the project in XCode and select test target
+2. Clik on the `Build Phases`
```
Comment (aliciakrebs) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:179]:
```suggestion
2. Click the `Build Phases`
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 4]
```diff
@@ -150,8 +159,87 @@ The next step is to trigger the test by following command:
...
+3. In the `Build Phases`, click on the `+`-sign
```
Comment (aliciakrebs) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:180]:
```suggestion
3. In the `Build Phases`, click the `+`-sign
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 5]
```diff
@@ -150,8 +159,87 @@ The next step is to trigger the test by following command:
...
+echo "$UNIQUE_CLASS_NAMES" > "$OUTPUT_FILE"
+
+# Print the class names and the path to the output file
+echo "Test class names:"
+echo "$UNIQUE_CLASS_NAMES"
+echo ""
+echo "Test class list has been created at: ${OUTPUT_FILE}"
+```
+For the steps also see the below image
+![add-build-script.png](/img/xcuitest/add-build-script.png)
```
Comment (aliciakrebs) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:221]: This screenshot may be a bit hard to follow. Can you send me the image without the callouts? I think I could help with this.
```suggestion
![add-build-script.png](/img/xcuitest/add-build-script.png)
```

Comment (tianfeng92) [reply docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:221]: This image is created by @wswebcreation so I don't have one without callouts. But I can help to take some screenshots from my side.

Comment (aliciakrebs) [reply docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:221]: thank you!

Comment (tianfeng92) [reply docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:221]: Just replaced these images. Can you help on that? Thanks!

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 6]
```diff
@@ -150,8 +159,87 @@ The next step is to trigger the test by following command:
 ```
 $ saucectl run
 ```
+
 `saucectl` uploads `DemoApp.ipa` and `DemoAppUITests-Runner.ipa` to Sauce VM and executes XCUITest. You can review test results on Sauce UI.
 
+#### Sharding XCUITest
+
+`saucectl` supports running tests in parallel. To tests in parallel, configure `saucectl` as follows. For more details, see [yaml configuration](./xcuitest.md#shard).
```
Comment (aliciakrebs) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:167]:
```suggestion
`saucectl` supports running tests in parallel. To tests in parallel, configure `saucectl` as follows. See [yaml configuration](./xcuitest.md#shard) for more details.
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 7]
```diff
@@ -150,8 +159,87 @@ The next step is to trigger the test by following command:
...
+# Write the unique class names to the output file
+echo "$UNIQUE_CLASS_NAMES" > "$OUTPUT_FILE"
+
+# Print the class names and the path to the output file
+echo "Test class names:"
+echo "$UNIQUE_CLASS_NAMES"
+echo ""
+echo "Test class list has been created at: ${OUTPUT_FILE}"
+```
+For the steps also see the below image
```
Comment (aliciakrebs) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:220]:
```suggestion
```

**Review by aliciakrebs — APPROVED** [review]

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md (hunk 5 above)
Comment (tianfeng92) [reply docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:220]: Thank you!

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 8]
```diff
@@ -150,8 +159,86 @@ The next step is to trigger the test by following command:
 ```
 $ saucectl run
 ```
+
 `saucectl` uploads `DemoApp.ipa` and `DemoAppUITests-Runner.ipa` to Sauce VM and executes XCUITest. You can review test results on Sauce UI.
 
+#### Sharding XCUITest
+
+`saucectl` supports running tests in parallel. To tests in parallel, configure `saucectl` as follows. See [yaml configuration](./xcuitest.md#shard) for more details.
```
Comment (alexplischke) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:167]:
```suggestion
`saucectl` supports running tests in parallel. To test in parallel, configure `saucectl` as follows. See [yaml configuration](./xcuitest.md#shard) for more details.
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 9]
```diff
@@ -150,8 +159,86 @@ The next step is to trigger the test by following command:
...
+  - name: "sharded suites"
+    shard: concurrency
+    testListFile: test_classes.txt
+```
+
+The `test_classes.txt` can be generated by following these steps.
+
+1. Open the project in XCode and select test target
+2. Click the `Build Phases`
+3. In the `Build Phases`, click the `+`-sign
```
Comment (alexplischke) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:180]:
```suggestion
3. In `Build Phases`, click the `+`-sign
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 10]
```diff
@@ -150,8 +159,86 @@ The next step is to trigger the test by following command:
...
+suites:
+  - name: "sharded suites"
+    shard: concurrency
+    testListFile: test_classes.txt
+```
+
+The `test_classes.txt` can be generated by following these steps.
+
+1. Open the project in XCode and select test target
+2. Click the `Build Phases`
```
Comment (alexplischke) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:179]:
```suggestion
2. Click `Build Phases`
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 11]
```diff
@@ -150,8 +159,86 @@ The next step is to trigger the test by following command:
...
+echo "Test class list has been created at: ${OUTPUT_FILE}"
+```
+![add-build-script.png](/img/xcuitest/add-build-script.png)
+
+5. The script can be triggered with
+    * XCODE by:
+        * Going to `Product` > `Build For` > `Testing`
+            ![trigger-build-script.png](/img/xcuitest/trigger-build-script.png)
+        * Or triggering a Testing build by using `Command + Shift + U`
+    * Or through running this command
```
Comment (alexplischke) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:227]: Flows better with the sentence above:

```suggestion
    * by XCODE:
        * Going to `Product` > `Build For` > `Testing`
            ![trigger-build-script.png](/img/xcuitest/trigger-build-script.png)
        * Or triggering a Testing build by using `Command + Shift + U`
    * by running this command
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md [hunk 12]
```diff
@@ -150,8 +159,86 @@ The next step is to trigger the test by following command:
...
+xcodebuild \
+  CODE_SIGN_IDENTITY="" \
+  CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO \
+  clean build-for-testing \
+  -workspace "{workspaceName}.xcworkspace" \
+  -scheme "{schemeName}" \
+  -sdk iphoneos
+```
+
+The generated `test_classes.txt` can be found here.
```
Comment (alexplischke) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md:238]: Colon probably better, as I assume that you are referring to the picture after this?
```suggestion
The generated `test_classes.txt` can be found here:
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest.md [hunk 13]
```diff
@@ -977,6 +977,33 @@ Instructs `saucectl` to run all classes for the suite _except_ those specified h
 
 ---
 
+### `shard`
+
+<p><small>| OPTIONAL | STRING |</small></p>
+
+When shard is configured as concurrency, saucectl automatically splits the tests by concurrency so that they can run in parallel.
```
Comment (alexplischke) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest.md:984]: Taken over from https://docs.saucelabs.com/web-apps/automated-testing/cypress/yaml/v1/#shard for consistency in writing.

```suggestion
When shard is configured as `concurrency`, saucectl automatically splits the tests into several groups (the number of groups is determined by the concurrency setting). Each group will then run as an individual job.
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest.md [hunk 14]
```diff
@@ -977,6 +977,33 @@ Instructs `saucectl` to run all classes for the suite _except_ those specified h
...
+    shard: concurrency
+```
+---
+
+### `testListFile`
+
+<p><small>| OPTIONAL | STRING |</small></p>
+
+The file containing tests is used in sharding by concurrency.
+
```
Comment (alexplischke) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest.md:1004]: Should the file format be described here? If not, where can the user find it? If the user searches for this field within the docs, they'll ask themselves that question and start looking around the page.

Comment (tianfeng92) [reply docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest.md:1004]: Ok. Will add it.

File: docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest.md [hunk 15]
```diff
@@ -977,6 +977,39 @@ Instructs `saucectl` to run all classes for the suite _except_ those specified h
...
+### `testListFile`
+
+<p><small>| OPTIONAL | STRING |</small></p>
+
+The file containing a list of tests is used in sharding by concurrency. It's a `txt` file and each line contains a test. Click [Sharding XCUITest introduction](./xcuitest-introduction.md#sharding-xcuitest) to see how to generate this file.
+
+```
+test1
+test2
+test3
```
Comment (alexplischke) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest.md:1008]: I know I listed this as an example myself in the Jira ticket, but for external purposes (and easier understanding), it'd be good to use a more descriptive approach. Since these are classes, and also methods, right?, it'd be good to list them as such here. Just more illustrative of how it's supposed to look like in the end.

**Review by alexplischke — APPROVED** [review]
